src/main/java/org/opentripplanner/graph_builder/linking/StreetSplitter.java

```diff
@@ -1,11 +1,834 @@
+     *
+     * StreetSplitter generates index on graph and splits destructively (used in transit splitter)
+     * @param graph
+     */
+    public StreetSplitter(Graph graph) {
+        this(graph, null, null);
+    }
+
+    /**
+     * Link all relevant vertices to the street network.
+     * This is to be called only during graph building in order to destructively split edges in order to create
+     * permanent edges.
+     */
+    public void linkAllStationsToGraph() {
+        for (Vertex v : graph.getVertices()) {
+            if (v instanceof TransitStop || v instanceof BikeRentalStationVertex || v instanceof BikeParkVertex)
+                if (!linkToClosestWalkableEdge(v, DESTRUCTIVE_SPLIT, false)) {
+                    if (v instanceof TransitStop)
+                        LOG.warn(graph.addBuilderAnnotation(new StopUnlinked((TransitStop) v)));
+                    else if (v instanceof BikeRentalStationVertex)
+                        LOG.warn(graph.addBuilderAnnotation(new BikeRentalStationUnlinked((BikeRentalStationVertex) v)));
+                    else if (v instanceof BikeParkVertex)
+                        LOG.warn(graph.addBuilderAnnotation(new BikeParkUnlinked((BikeParkVertex) v)));
+                };
+        }
+    }
+
+    /**
+     * Link this vertex into the graph to the closest walkable edge
+     * @param vertex The vertex to be linked.
+     * @param destructiveSplitting If true splitting is permanent (Used when linking transit stops etc.) when
+     *                             false the original edge is kept in the graph after creating split edges.
+     * @param createSemiPermanentEdges If true, semi-permanent edges and a semi-permanent splitter vertex is created.
+     *                                 These vertices and edges are intended to be kept in the graph and linked to
+     *                                 during subsequent requests, but might be removed at a later time. For example a
+     *                                 floating bike rental may need to be removed from the graph once the data is
+     *                                 updated to show that the floating bike rental no longer exists or if its
+     *                                 location has significantly changed.
+     */
+    public boolean linkToClosestWalkableEdge(
+        Vertex vertex,
+        final boolean destructiveSplitting,
+        boolean createSemiPermanentEdges
+    ) {
+        return linkToGraph(vertex, TraverseMode.WALK, null, destructiveSplitting, createSemiPermanentEdges);
+    }
+
+    public boolean linkToGraph(
+        Vertex vertex,
+        TraverseMode traverseMode,
+        RoutingRequest options,
+        final boolean destructiveSplitting,
+        boolean createSemiPermanentEdges
+    ) {
+        final TraverseModeSet traverseModeSet = new TraverseModeSet(traverseMode);
+        if (traverseMode == TraverseMode.BICYCLE) {
+            traverseModeSet.setWalk(true);
+        }
```

Comment:
This setting of walk mode for bicycle... should it also be doing the same for micromobility? I think a comment would be helpful here as well.

Reply:
Yes, I think so.